sqlite3/lib/src/wasm/file_system.dart

```diff
@@ -63,23 +72,40 @@ abstract class FileSystem {
 /// An exception thrown by a [FileSystem] implementation.
 class FileSystemException implements Exception {
   final int errorCode;
+  final String message;
 
-  FileSystemException([this.errorCode = SqlError.SQLITE_ERROR]);
+  FileSystemException(
+      [this.errorCode = SqlError.SQLITE_ERROR, this.message = 'SQLITE_ERROR']);
 
   @override
   String toString() {
-    return 'FileSystemException($errorCode)';
+    return 'FileSystemException: ($errorCode) $message';
   }
 }
 
+/// An exception thrown by a [FileSystem] implementation when try to access file
+/// outside of persistence root
+class FileSystemAccessException extends FileSystemException {
```

Comment:
IMO we don't need a separate class for this, we can just throw a `FileSystemException` with that error code and message directly. Or do you see a reason for users to explicitly catch this type?

Reply:
I thought it is a more verbose exception to path validation and different from "regular" IO exceptions. But of course I can remove it. By the way, if current implementation will be replaced, this wouldn't need anymore.